projects/ngrx-auto-entity/src/index.ts

```diff
@@ -30,109 +53,96 @@ export { fromEntityActions } from './lib/actions/action-operators';
+export { Replace, ReplaceFailure, ReplaceSuccess } from './lib/actions/replace-actions';
+export { DeleteMany, DeleteManyFailure, DeleteManySuccess } from './lib/actions/delete-actions';
+export { Delete, DeleteFailure, DeleteSuccess } from './lib/actions/delete-actions';
 export {
-  Load,
-  LoadFailure,
-  LoadSuccess,
-  LoadAll,
-  LoadAllFailure,
-  LoadAllSuccess,
-  LoadMany,
-  LoadManyFailure,
-  LoadManySuccess,
-  LoadPage,
-  LoadPageFailure,
-  LoadPageSuccess,
-  LoadRange,
-  LoadRangeFailure,
-  LoadRangeSuccess,
-  CreateFailure,
-  CreateSuccess,
-  Create,
-  CreateMany,
-  CreateManyFailure,
-  CreateManySuccess,
-  Update,
-  UpdateSuccess,
-  UpdateFailure,
-  UpdateMany,
-  UpdateManySuccess,
-  UpdateManyFailure,
-  Upsert,
-  UpsertSuccess,
-  UpsertFailure,
-  UpsertMany,
-  UpsertManySuccess,
-  UpsertManyFailure,
-  Replace,
-  ReplaceFailure,
-  ReplaceSuccess,
-  ReplaceMany,
-  ReplaceManyFailure,
-  ReplaceManySuccess,
-  Delete,
-  DeleteFailure,
-  DeleteSuccess,
-  DeleteByKey,
-  DeleteByKeyFailure,
-  DeleteByKeySuccess,
-  DeleteMany,
-  DeleteManyFailure,
-  DeleteManySuccess,
   DeleteManyByKeys,
   DeleteManyByKeysFailure,
   DeleteManyByKeysSuccess,
+  DeleteByKeyFailure,
+  DeleteByKeySuccess,
+  DeleteByKey
+} from './lib/actions/delete-by-key-actions';
+
+export {
   Select,
   SelectByKey,
-  SelectMany,
-  SelectMore,
-  SelectManyByKeys,
-  SelectMoreByKeys,
   Selected,
   SelectedMany,
-  Deselect,
-  DeselectMany,
-  DeselectManyByKeys,
-  DeselectAll,
-  Deselected,
-  Edit,
-  EditByKey,
-  Edited,
-  EditedByKey,
-  Change,
-  Changed,
-  EndEdit,
-  EditEnded,
-  Clear
-} from './lib/actions/actions';
+  SelectMany,
+  SelectManyByKeys,
+  SelectMore,
+  SelectMoreByKeys
+} from './lib/actions/selection-actions';
+export { Deselected, DeselectAll, DeselectManyByKeys, DeselectMany, Deselect } from './lib/actions/deselection-actions';
+export { EditEnded, EndEdit, Changed, Change, EditedByKey, Edited, EditByKey, Edit } from './lib/actions/edit-actions';
+export { Clear } from './lib/actions/actions';
+
+/*
+ * Decorators
+ */
+export { Entity } from './lib/decorators/entity-decorator';
+export { Key } from './lib/decorators/key-decorator';
 
+/*
+ * Entity Metadata and Management
+ */
 export { ENTITY_OPTS_PROP } from './lib/decorators/entity-tokens';
-export { IEffectExcept, IEntityOptions, IEntityTransformer, Entity } from './lib/decorators/entity';
+export { IEffectExcept, IEntityOptions, IEntityTransformer, EntityAge } from './lib/decorators/entity-options';
 export { IEffectExclusions } from './lib/decorators/effect-exclusions';
 export { curd, loads, extra, all, matching, except } from './lib/decorators/effect-exclusion-utils';
+
+/*
+ * Entity Metadata Utilities
+ */
 export { makeEntity } from './lib/util/make-entity';
 export {
   nameOfEntity,
   pluralNameOfEntity,
   uriNameOfEntity,
+  stateNameOfEntity,
   entityComparer,
   entityTransforms
```

Comment:
Need to include `entityMaxAge`, which might better be renamed to `defaultMaxAge`